pkg/dockerregistry/server/auth.go

```diff
@@ -415,17 +449,37 @@ func verifyPruneAccess(ctx context.Context, client client.SubjectAccessReviews)
 			Resource: "images",
 		},
 	}
-	response, err := client.SubjectAccessReviews().Create(&sar)
-	if err != nil {
-		context.GetLogger(ctx).Errorf("OpenShift client error: %s", err)
-		if kerrors.IsUnauthorized(err) || kerrors.IsForbidden(err) {
+	return withExponentialBackoff(func() error {
+		response, err := client.SubjectAccessReviews().Create(&sar)
+		if err != nil {
+			context.GetLogger(ctx).Errorf("OpenShift client error: %s", err)
+			return err
+		}
+		if !response.Allowed {
+			context.GetLogger(ctx).Errorf("OpenShift access denied: %s", response.Reason)
 			return ErrOpenShiftAccessDenied
 		}
-		return err
+		return nil
+	})
+}
+
+func withExponentialBackoff(handler func() error) (err error) {
```

Comment:
It would be more correct to define your constant and then just call wait.Exponential backoff.  This construct hides what you are actually doing.  

```
var backoff =

func () ... {
  ...
  wait.ExponentialBackoff(backoff, verifyError(function))
}

```

would be more idiomatic.